src/v/archival/ntp_archiver_service.cc

```diff
@@ -107,14 +107,26 @@ ss::future<> ntp_archiver::start() {
     return ss::now();
 }
 
+void ntp_archiver::notify_leadership(std::optional<model::node_id> leader_id) {
+    if (leader_id && *leader_id == _parent.raft()->self().id()) {
+        _leader_cond.signal();
+    }
+}
+
 ss::future<> ntp_archiver::outer_upload_loop() {
     if (!_probe) {
         _probe.emplace(_conf->ntp_metrics_disabled, _ntp);
     }
 
     while (!_as.abort_requested()) {
         if (!_parent.is_elected_leader()) {
-            co_await ss::sleep_abortable(_conf->reconciliation_interval, _as);
+            try {
+                co_await _leader_cond.wait();
+            } catch (const ss::broken_condition_variable&) {
+                // stop() was called
+                break;
+            }
+
             continue;
```

Comment:
The `continue` is not required anymore. We know we are the leader, so we can proceed and start the loop.

Reply:
True... I wrote this intending to use leader_cond for any leadership change, but ended up only signalling it when we are leader, so can remove this.

Reply:
Fixed.